src/widget/mod.rs

```diff
@@ -1,16 +1,17 @@
 pub mod container;
 pub mod text;
 
+mod intrinsic_size;
+
+pub use intrinsic_size::IntrinsicSize;
+
 use alloc::boxed::Box;
 use embedded_graphics::prelude::*;
 
 pub trait Widget<Display: DrawTarget> {
-    fn draw(
-        &self,
-        display: &mut Display,
-        origin: Point,
-        max_size: Size,
-    ) -> Result<Size, Display::Error>;
+    fn intrinsic_size(&self) -> IntrinsicSize;
+
+    fn draw(&self, display: &mut Display, origin: Point, size: Size) -> Result<(), Display::Error>;
```

Comment:
The idea I'm going with so far is that the parent of a widget (the caller of its `draw` function) is responsible for deciding the exact size of a widget. It can ask the widget what size it'd prefer using `intrinsic_size` to make that decision.